### PR TITLE
Added workflow to publish package on npm

### PR DESCRIPTION
**Pull request by serhii-temchenko**
In this pull request, a new GitHub workflow has been added to publish a new package on NPM after merging or pushing changes to the `main` branch.

Closes issue #16. 

:warning: ~~To make it work - `NPM_AUTH_TOKEN` secret should be added as a repo secret where value is an issued `publication` token from NPM.~~ :heavy_check_mark: Added.

### CODE REVIEW

**Comment by serhii-temchenko** [comment]
This PR currently is blocked by PR #20.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "dev": "turbo run dev",
     "lint": "turbo run lint",
     "format": "prettier --write \"**/*.{ts,tsx,md}\"",
-    "publish-packages": "turbo run build lint test && changeset version && changeset publish"
+    "release": "turbo run build lint && changeset publish"
```
Comment (nuel77) [inline package.json:9]: for release, we can just have  `changeset publish` . the run and build are done on every pr before merge so no need to do it here again.

Comment (serhii-temchenko) [reply package.json]: Updated.

**Review by nuel77 — APPROVED** [review]

**Review by minikas — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        with:
+          node-version: '20.x'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Creating .npmrc
+        run: |
+          cat << EOF > "$HOME/.npmrc"
```
Comment (nuel77) [inline .github/workflows/publish.yml:34]: this is automatically created by the changeset github action?

**Review by nuel77 — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (serhii-temchenko) [reply .github/workflows/publish.yml]: Agree, added it before actual usage of changeset action and didn't removed.
Now I removed this useless action on recent commit. Please check.

**Pull request merged**